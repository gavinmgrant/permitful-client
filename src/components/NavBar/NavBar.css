nav {
    position: sticky;
    top: 0;
    left: 0;
    right: 0;
    color: #18181E;
    background-color: #DEC79B;
    margin: 0;
    padding: 15px;
    z-index: 20;
    display: flex;
    justify-content: space-between;
}

.nav-bar:hover {
    border-bottom: 2px solid #18181E;
}

ul {
    display: inline;
    list-style: none;
    list-style-type: none;
    padding: 0;
    margin: 2px 0 0 0;
}

li {
    display: inline;
    padding: 5px;
}

.logo {
    text-decoration: none;
    padding: 10px;
    font-family: 'Quicksand', sans-serif;
    color: #18181E;
}

.logo:visited {
    color: #18181E;
}

.nav-bar,
.nav-bar:visited {
    color: #18181E;
    margin-left: 5px;
}

@media screen and (min-width: 1024px) {
    nav {
        padding: 20px;
    }
    
    li {
        padding: 10px;
    }
}