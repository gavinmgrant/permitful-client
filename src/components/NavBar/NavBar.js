import React, { Component } from 'react';
import { Link } from 'react-router-dom';
import TokenService from '../../services/token-service';
import IdleService from '../../services/idle-service';
import './NavBar.css';

export default class NavBar extends Component {
    handleLogoutClick = () => {
        TokenService.clearAuthToken();
        window.history.back();
        // when logging out, clear the callbacks to the refresh api and idle auto logout
        TokenService.clearCallbackBeforeExpiry()
        IdleService.unRegisterIdleResets()
    }

    // renders the links for when users are signed in
    renderLogoutLink() {
        return (
            <div className='navbar-logged-in'>
                <ul>
                    <Link 
                        to='/favorites' 
                        style={{ textDecoration: 'none' }}
                    >
                        <li className='nav-bar'>Favorites</li>
                    </Link>
                    <Link
                        onClick={this.handleLogoutClick}
                        to='/'
                        style={{ textDecoration: 'none' }}
                    >
                        <li className='nav-bar'>Log Out</li>
                    </Link>
                </ul>
            </div>
        )
    }

    // renders the links for when no user is signed in
    renderLoginLink() {
        return (
            <div className='navbar-not-logged-in'>
                <ul>
                    <Link 
                        to='/register' 
                        style={{ textDecoration: 'none' }}
                    >
                        <li className='nav-bar'>Register</li>
                    </Link>
                    <Link 
                        to='/login' 
                        style={{ textDecoration: 'none' }}
                    >
                        <li className='nav-bar'>Log In</li>
                    </Link>
                </ul>
            </div>
        )
    }

    render() {
        return (
            <nav role="navigation">
                <h3>
                    <Link to='/' className='logo'>
                        Permitful
                    </Link>
                </h3>
                {TokenService.hasAuthToken()
                    ? this.renderLogoutLink()
                    : this.renderLoginLink()}
            </nav>
        )
    }
}