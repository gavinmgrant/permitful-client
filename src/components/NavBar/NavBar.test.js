import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import NavBar from './NavBar';

describe(`NavBar component`, () => {
    it('renders a navbar by default', () => {
      const wrapper = shallow(<NavBar />)
      expect(toJson(wrapper)).toMatchSnapshot()
    })
});