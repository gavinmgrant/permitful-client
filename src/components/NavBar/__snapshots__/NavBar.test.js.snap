// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar component renders a navbar by default 1`] = `
<nav
  role="navigation"
>
  <h3>
    <Link
      className="logo"
      to="/"
    >
      Permitful
    </Link>
  </h3>
  <div
    className="navbar-not-logged-in"
  >
    <ul>
      <Link
        style={
          Object {
            "textDecoration": "none",
          }
        }
        to="/register"
      >
        <li
          className="nav-bar"
        >
          Register
        </li>
      </Link>
      <Link
        style={
          Object {
            "textDecoration": "none",
          }
        }
        to="/login"
      >
        <li
          className="nav-bar"
        >
          Log In
        </li>
      </Link>
    </ul>
  </div>
</nav>
`;
